pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +76,60 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+//
+// Precedence of failures:
+// 1. A build config's input points to an image stream that does not exist
+// 2. A build config's input uses an image stream tag reference in an existing image stream, but no images within the image stream have that tag assigned
+// 3. A build config's input uses an image stream image reference in an exisiting image stream, but no images within the image stream have the supplied image hexadecimal ID
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
```

Comment:
`istNode` is a bad name now, right? This could be an IST or an ISI, right?  Should the content of this loops be split based on the switch/case for type?